src/main/java/com/example/onlinebookstore/service/shopping/cart/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,101 @@
+package com.example.onlinebookstore.service.shopping.cart;
+
+import com.example.onlinebookstore.dto.cart.items.CreateCartItemRequestDto;
+import com.example.onlinebookstore.dto.shopping.cart.CartItemUpdateRequestDto;
+import com.example.onlinebookstore.dto.shopping.cart.ShoppingCartDto;
+import com.example.onlinebookstore.exception.EntityNotFoundException;
+import com.example.onlinebookstore.mapper.ShoppingCartMapper;
+import com.example.onlinebookstore.model.Book;
+import com.example.onlinebookstore.model.CartItem;
+import com.example.onlinebookstore.model.ShoppingCart;
+import com.example.onlinebookstore.model.User;
+import com.example.onlinebookstore.repository.BookRepository;
+import com.example.onlinebookstore.repository.CartItemRepository;
+import com.example.onlinebookstore.repository.ShoppingCartRepository;
+import com.example.onlinebookstore.repository.UserRepository;
+import jakarta.transaction.Transactional;
+import java.util.HashSet;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final UserRepository userRepository;
+    private final BookRepository bookRepository;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    @Transactional
+    public ShoppingCartDto findById(Long id) {
+        return shoppingCartMapper.toDto(shoppingCartRepository.findById(id)
+                .orElseThrow(() -> new RuntimeException(
+                        "Cannot find shoppingCart by id " + id)
+                )
+        );
+    }
+
+    @Override
+    @Transactional
+    public ShoppingCartDto addToCart(Long id, CreateCartItemRequestDto requestDto) {
+        User user = userRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Cannot find user")
+        );
+        ShoppingCart shoppingCart = shoppingCartRepository.findById(id)
+                .orElseGet(() -> {
+                    ShoppingCart shoppingCartNew = new ShoppingCart();
+                    shoppingCartNew.setUser(user);
+                    shoppingCartNew.setCartItems(new HashSet<>());
+                    return shoppingCartRepository.save(shoppingCartNew);
+                }
+        );
+        Book book = bookRepository.findById(requestDto.bookId()).orElseThrow(
+                () -> new EntityNotFoundException("Cannot find book")
+        );
+        CartItem cartItem = new CartItem();
+        cartItem.setBook(book);
+        cartItem.setShoppingCart(shoppingCart);
+        cartItem.setQuantity(requestDto.quantity());
+        shoppingCart.getCartItems().add(cartItem);
+        cartItemRepository.save(cartItem);
```

Comment:
Few comments:
1. What will be if you add item with the same bookId that exist in cart? I guess it add a new entity, but you need update quantity if existing.
2. You can replace cartItem.setShoppingCart(shoppingCart); and shoppingCart.getCartItems().add(cartItem); by helper method in ShoppingCart entity in the same way that you did with remove.
3. And why you don't use mapper to map requestDto to CartItem?